internal/transportmock/registry.go

```diff
@@ -0,0 +1,73 @@
+package transportmock
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"sync"
+	"testing"
+)
+
+type Registry struct {
+	mu       sync.Mutex
+	stubs    []*stub
+	requests []*http.Request
```

Comment:
I purposefully un-exported these and added a `Requests()` accessor function. It felt smart to move this direction incase we want to change the underlying implementation.